### PR TITLE
Specify list/map capacity when initializing where possible

### PR DESCRIPTION
**Pull request by mina-asham**
- This really depends on the use case, but in some cases the array/map resizing can be slightly costly, there is effectively no downside setting the initial capacity for a collection if we know for sure its final size

**Description**: Minor performance improvement by guaranteeing arrays don't resize unnecessarily

**Motivation and Context**
- Why is this change required? What problem does it solve?
  - No downsides, upside possible performance improvements in some cases
- If it fixes an open issue, please link to the issue here.
  - N/A

### CODE REVIEW

**Comment by faxu** [comment]
@Craigacp can you help review this?

File: java/src/main/java/ai/onnxruntime/MapUtil.java [hunk 1]
```diff
@@ -0,0 +1,14 @@
+package ai.onnxruntime;
+
+final class MapUtil {
+
+  private MapUtil() {
+    throw new UnsupportedOperationException(
+        "MapUtil is a utility class and should not be instantiated");
+  }
+
+  static int capacityFromSize(int size) {
```
Comment (Craigacp) [inline java/src/main/java/ai/onnxruntime/MapUtil.java:10]: Can you move this method onto the existing `OrtUtil` utility class?

**Review by Craigacp — CHANGES_REQUESTED** [review]
I don't think it's worth adding a whole class for the single method, so move that to `OrtUtil`, and then otherwise that's fine.

I'm kinda surprised that this was an issue given the size of the objects that are usually returned by those methods (where the default capacity is probably bigger than the sized one anyway), and the relative amount of computation in Java vs the ORT native code, but it doesn't really make much difference to include it.

**Comment by mina-asham** [comment]
> I don't think it's worth adding a whole class for the single method, so move that to `OrtUtil`, and then otherwise that's fine.
> 
> I'm kinda surprised that this was an issue given the size of the objects that are usually returned by those methods (where the default capacity is probably bigger than the sized one anyway), and the relative amount of computation in Java vs the ORT native code, but it doesn't really make much difference to include it.

I can confirm that most of the time in our OrtSession is not coming from native code, from JFR I also see a good chunk is coming from there, I don't expect the change to be noticeable for *most* cases but I expect an enhancement for ours and the rest should be same or better after this.

File: java/src/test/java/ai/onnxruntime/OrtUtilTest.java [hunk 2]
```diff
@@ -0,0 +1,18 @@
+package ai.onnxruntime;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+import org.junit.jupiter.api.Test;
+
+class OrtUtilTest {
+
+  @Test
+  void capacityFromSize() {
```
[the remark this answers is not in the document]
Comment (mina-asham) [reply java/src/test/java/ai/onnxruntime/OrtUtilTest.java]: Done

**Comment by faxu** [comment]
/azp run onnxruntime-python-checks-ci-pipeline, orttraining-amd-gpu-ci-pipeline, orttraining-linux-ci-pipeline, orttraining-linux-gpu-ci-pipeline, orttraining-ortmodule-distributed, 
ONNX Runtime Web CI Pipeline, onnxruntime-binary-size-checks-ci-pipeline

**Comment by mina-asham** [comment]
@faxu any blockers for merging this one?

**Comment by faxu** [comment]
@pranavsharma any concerns on this PR?

**Review by pranavsharma — APPROVED** [review]

**Pull request merged**